packages/compose/src/hooks/use-constrained-tabbing/index.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { TAB } from '@wordpress/keycodes';
+import { focus } from '@wordpress/dom';
+
+/**
+ * In Dialogs/modals, the tabbing must be constrained to the content of
+ * the wrapper element. This hook adds the behavior to the returned ref.
+ *
+ * @return {Function} Element Ref.
+ *
+ * @example
+ * ```js
+ * import { useConstrainedTabbing } from '@wordpress/compose';
+ *
+ * const ConstrainedTabbingExample = () => {
+ *     const constrainedTabbingRef = useConstrainedTabbing()
+ *     return (
+ *         <div ref={ constrainedTabbingRef }>
+ *             <Button />
+ *             <Button />
+ *         </div>
+ *     );
+ * }
+ * ```
+ */
+function useConstrainedTabbing() {
+	const ref = useCallback( ( node ) => {
+		if ( ! node ) {
+			return;
+		}
+		node.addEventListener( 'keydown', ( event ) => {
```

Comment:
A clean up is missing, which would be easier to achieve with `useEffect`.

Reply:
no, we don't need a cleanup here because it's  a ref callback meaning  if we need to cleanup it means the node is already  "gone" which  means there's nothing to clean up.